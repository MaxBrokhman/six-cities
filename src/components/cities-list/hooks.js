import {setCity} from '../../reducer/actions';

export const useCities = (offers, dispatch) => {
  const uniqueCities = [];
  for (let {city} of offers) {
    if (uniqueCities.indexOf(city) === -1) {
      uniqueCities.push(city);
    }
  }

  const clickHandler = (city) => (evt) => {
    evt.preventDefault();
    setCity(city, dispatch);
  };

  return {
    cities: uniqueCities,
    clickHandler,
  };
};
