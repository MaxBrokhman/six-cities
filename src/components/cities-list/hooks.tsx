import {SyntheticEvent} from 'react';

import {setCity} from '../../reducer/actions';
import {TOffer, TDispatch} from '../../reducer/types';

type TUseCities = {
  cities: Array<string>;
  clickHandler: (city: string) => (evt: SyntheticEvent) => void;
}

export const useCities = (offers: Array<TOffer>, dispatch: TDispatch): TUseCities => {
  const uniqueCities = [];
  for (const {city} of offers) {
    if (uniqueCities.indexOf(city.name) === -1) {
      uniqueCities.push(city.name);
    }
  }

  const clickHandler = (city: string) => (evt: SyntheticEvent): void => {
    evt.preventDefault();
    setCity(city, dispatch);
  };

  return {
    cities: uniqueCities,
    clickHandler,
  };
};
