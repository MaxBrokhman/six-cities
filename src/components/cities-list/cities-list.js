import React from 'react';

import {useAppContext} from '../../reducer/reducer';
import {useCities} from './hooks';

export const CitiesList = () => {
  const {state, dispatch} = useAppContext();
  const {cities, clickHandler} = useCities(state.offersList, dispatch);
  return (
    <ul className="locations__list tabs__list">
      {
        cities.map((city) => (
          <li className="locations__item" key={city}>
            <a
              className="locations__item-link tabs__item"
              href="#"
              onClick={clickHandler(city)}
            >
              <span>{city}</span>
            </a>
          </li>
        ))
      }
    </ul>
  );
};
