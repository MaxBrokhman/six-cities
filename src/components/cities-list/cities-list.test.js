import React from 'react';
import renderer from 'react-test-renderer';

import {CitiesList} from './cities-list';

it(`Renders CitiesList component correctly`, () => {
  const tree = renderer
                .create(<CitiesList />)
                .toJSON();
  expect(tree).toMatchSnapshot();
});
