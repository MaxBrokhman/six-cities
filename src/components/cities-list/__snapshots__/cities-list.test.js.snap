// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders CitiesList component correctly 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        amsterdam
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        berlin
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        paris
      </span>
    </a>
  </li>
</ul>
`;
